plugins/inputs/redis/redis.go

```diff
@@ -686,3 +694,18 @@ func coerceType(value interface{}, typ reflect.Type) reflect.Value {
 	}
 	return reflect.ValueOf(value)
 }
+
+func (r *Redis) Start(acc telegraf.Accumulator) error {
+	r.acc = acc
+	return nil
+}
+
+//Stop close the client through ServiceInput interface Start/Stop methods impl.
+func (r *Redis) Stop() {
+	for _, c := range r.clients {
+		err := c.Close()
+		if err != nil {
+			r.acc.AddError(fmt.Errorf("error closing client: %s", err.Error()))
```

Comment:
Can you maybe just log the error using
```suggestion
			r.Log.Errorf("error closing client: %v", err)
```
instead of mis-using the accumulator?
If no logger exists yet, please just add the following line to the plugin struct
```golang
	Log telegraf.Logger `toml:"-"`
```

Reply:
Thanks.